module/module.go

```diff
@@ -268,19 +253,26 @@ func (m *Module) GetParentResource(ctx context.Context, name resource.Name) (res
 }
 
 func (m *Module) connectParent(ctx context.Context) error {
-	m.mu.Lock()
-	defer m.mu.Unlock()
-	if m.parent == nil {
-		if err := CheckSocketOwner(m.parentAddr); err != nil {
-			return err
-		}
-		// TODO(PRODUCT-343): add session support to modules
-		rc, err := client.New(ctx, "unix://"+m.parentAddr, m.logger, client.WithDisableSessions())
-		if err != nil {
-			return err
-		}
-		m.parent = rc
+	// If parent connection has already been made, do not make another one.
```

Comment:
Found this issue from `TestModuleFunctions` which sends two `ReadyRequest`s concurrently. If an `rdk` were to retry sending a `ReadyRequest` to a module for any reason, I think we could feasibly make a second connection back to the parent and leak the first, so I just disallowed the setting of `parent` more than once.

Reply:
Can you add this information to the documentation? I'm fine keeping the nil-check for a test. Just want a reader to know `connectParent` will be called once per-ReadyRequest. And serving multiple `ReadyRequest`s is only expected to happen in tests.

Reply:
Yep! Can add more of my original comment to the inline comment.